api/src/model/request/play_asset_pack_fetch_request.gd

```diff
@@ -0,0 +1,130 @@
+var _stub_play_asset_pack_manager
+
+func _init(pack_name):
+	_pack_name = pack_name
+	
+	# _state will be defaulted with status of UNKNOWN
+	var default_pack_dict = {
+		PlayAssetPackState._NAME_KEY: pack_name, 
+		PlayAssetPackState._STATUS_KEY: PlayAssetPackManager.AssetPackStatus.UNKNOWN, 
+		PlayAssetPackState._ERROR_CODE_KEY: PlayAssetPackManager.AssetPackErrorCode.NO_ERROR,
+		PlayAssetPackState._BYTES_DOWNLOADED_KEY: 0,
+		PlayAssetPackState._TOTAL_BYTES_TO_DOWNLOAD_KEY: 0,
+		PlayAssetPackState._TRANSFER_PROGRESS_PERCENTAGE_KEY: 0
+	}
+	_state = PlayAssetPackState.new(default_pack_dict)
+	
+
+# -----------------------------------------------------------------------------
+# Returns the requested asset pack's name.
+# -----------------------------------------------------------------------------
+func get_pack_name() -> String:
+	return _pack_name
+
+# -----------------------------------------------------------------------------
+# Returns boolean indicating Request succeeded/failed.
+# -----------------------------------------------------------------------------
+func get_did_succeed() -> bool:
+	return _did_succeed
+
+# -----------------------------------------------------------------------------
+# Returns the most up-to-date PlayAssetPackState object, null if Request failed.
+# -----------------------------------------------------------------------------
+func get_state() -> PlayAssetPackState:
+	return _state
+
+# -----------------------------------------------------------------------------
+# Returns a PlayAssetPackException if exception occurred.
+# -----------------------------------------------------------------------------
+func get_error() -> PlayAssetPackException:
+	return _error
+
+# -----------------------------------------------------------------------------
+# Callback functions handling signals emitted from the plugin.
+# -----------------------------------------------------------------------------
+func _on_fetch_success(result: Dictionary):
+	# Since fetch() in plugin returns a PlayAssetPackStates Dictionary, we need to extract
+	# the PlayAssetPackState within.
+	var fetch_asset_pack_states_dict = PlayAssetPackStates.new(result).get_pack_states()
+	
+	if fetch_asset_pack_states_dict.has(_pack_name):
+		_on_state_updated(fetch_asset_pack_states_dict[_pack_name].to_dict())
+		
+		# emit non-duplicating state_updated signal
+		if _stub_play_asset_pack_manager == null:
+			PlayAssetPackManager._forward_high_level_state_updated_signal(_pack_name, _state.to_dict())
+		else:
+			_stub_play_asset_pack_manager._forward_high_level_state_updated_signal(_pack_name, _state.to_dict())	
+	else:
+		# Although we received a fetchSuccess signal, the result field does not contain
+		# needed AssetPackState dictionary. Hence update _state's error_code to INVALID_REQUEST
+		# and emit and request_completed signal.
+		_did_succeed = false
+		_state._error_code = PlayAssetPackManager.AssetPackErrorCode.INVALID_REQUEST
```

Comment:
I think we talked about this being changed to INTERNAL_ERROR? The thinking was that Fetch() will handle INVALID_REQUEST by calling onFailure on the task object. Fetch() should never call onSuccess with an assetPackStates object that didn't include this state. We can use INTERNAL_ERROR to indicate that there's nothing the user can do to remedy this error, and we have no idea what caused it.

Reply:
nvm I see you updated this before I posted this comment.